### PR TITLE
Update alias for NatGateway and NatGatewayId

### PR DESCRIPTION
**Pull request by khannarheams**
## Description

Update Alias for NatGateway and NatGatewayId

## Checklist

- [ ] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/master/CONTRIBUTING.md#updating-the-change-log)
- [ ] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (Microsoft internal partners only)
    - [ ] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/help-generation.md)

### CODE REVIEW

File: src/Network/Network/help/Remove-AzNatGateway.md [hunk 1]
```diff
@@ -38,8 +38,8 @@ Remove Nat Gateway Resource
 ### Example 1
 ```powershell
 PS C:> $nat = Get-AzNatGateway -ResourceGroupName "natgateway_test" -Name "nat_gateway"
-PS C:> Remove-AzNatGateway -InputObject $nat
```
Comment (niander) [inline src/Network/Network/help/Remove-AzNatGateway.md]: The cmdlets that have NatGateway as a noun should continue to have `-InputObject` and `-ResourceId`. Please revert the changes in examples and by just regenerating the help files the syntax should also be updated.

Comment (niander) [reply src/Network/Network/help/Remove-AzNatGateway.md]: Do this everywhere else applicable

Comment (khannarheams) [reply src/Network/Network/help/Remove-AzNatGateway.md]: Updated this since we want to represent alias in the examples

Comment (niander) [reply src/Network/Network/help/Remove-AzNatGateway.md]: You can update the examples to use the alias instead of the original parameter name but because the syntax and parameter information are always replaced when the help files are regenerated we cannot update them manually.

In addition, the cmdlets that have NatGateway in the noun name should continue to follow ours guidelines using `-ResourceId` and `-InputObject`. Since the cmdlets have NatGateway in the noun it should be clear for users which ResourceId or InputObject these parameters refer to. However, as we talked before you can still add aliases for these parameters, nevertheless.

Comment (niander) [reply src/Network/Network/help/Remove-AzNatGateway.md]: Don't forget to regenerate all help files, please.

Comment (khannarheams) [reply src/Network/Network/help/Remove-AzNatGateway.md]: @niander regenerated help files, please review.

File: src/Network/Network/help/Get-AzVirtualNetworkSubnetConfig.md [hunk 2]
```diff
@@ -83,7 +83,7 @@ Accept wildcard characters: False
 ```
 
 ### CommonParameters
-This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see [about_CommonParameters](http://go.microsoft.com/fwlink/?LinkID=113216).
+This cmdlet supports the common parameters: -Debug, -ErrorAction, -ErrorVariable, -InformationAction, -InformationVariable, -OutVariable, -OutBuffer, -PipelineVariable, -Verbose, -WarningAction, and -WarningVariable. For more information, see about_CommonParameters (http://go.microsoft.com/fwlink/?LinkID=113216).
```
Comment (niander) [inline src/Network/Network/help/Get-AzVirtualNetworkSubnetConfig.md]: Please update your `platyPS` to its latest version. This would avoid this incorrect change.

File: src/Network/Network/help/New-AzVirtualNetworkSubnetConfig.md [hunk 3]
```diff
@@ -68,7 +69,7 @@ This example creates two new subnet configurations using the
 Specifies a range of IP addresses for a subnet configuration.
 
 ```yaml
-Type: System.String[]
+Type: String[]
```
Comment (niander) [inline src/Network/Network/help/New-AzVirtualNetworkSubnetConfig.md]: When you do `Update-MarkdownHelpModule` use also the parameter `-UseFullTypeName`. This would keep the use of full type names.

Comment (khannarheams) [reply src/Network/Network/help/New-AzVirtualNetworkSubnetConfig.md]: updated, please review

File: src/Network/Network/help/Get-AzVirtualNetworkSubnetConfig.md (hunk 2 above)
Comment (khannarheams) [reply src/Network/Network/help/Get-AzVirtualNetworkSubnetConfig.md]: updated, please review @niander

File: src/Network/Network/help/New-AzVirtualNetworkSubnetConfig.md [hunk 4]
```diff
@@ -109,6 +119,21 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -InputObject
+Specifies the nat gateway associated with the subnet configuration
+
+```yaml
+Type: Microsoft.Azure.Commands.Network.Models.PSNatGateway
+Parameter Sets: SetByResource
+Aliases:
```
Comment (niander) [inline src/Network/Network/help/New-AzVirtualNetworkSubnetConfig.md]: The aliases are not being added. Are you sure that you built the latest changes and ran the update command using these latest changes? If still the aliases are not being added try to completely delete the help file before regenerating.

File: src/Network/Network/help/Set-AzVirtualNetworkSubnetConfig.md [hunk 5]
```diff
@@ -81,6 +81,20 @@ This example creates a resource group with one virtual network containing just o
     security group. The Set-AzVirtualNetwork cmdlet is then called to write the modified 
     state back to the service.
 
+### 3: Attach a Nat Gateway to a subnet
```
Comment (EvgenyAgafonchikov) [inline src/Network/Network/help/Set-AzVirtualNetworkSubnetConfig.md]: This help file is for Set-AzVirtualNetworkSubnetConfig and there is no `Set-` call in the example, please modify.

File: src/Network/Network/help/Add-AzVirtualNetworkSubnetConfig.md [hunk 6]
```diff
@@ -105,6 +105,19 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -InputObject
```
Comment (number213) [inline src/Network/Network/help/Add-AzVirtualNetworkSubnetConfig.md]: Please add description for this parameter

File: src/Network/Network/help/Add-AzVirtualNetworkSubnetConfig.md [hunk 7]
```diff
@@ -151,6 +164,21 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -ResourceId
+NatGatewayId
```
Comment (number213) [inline src/Network/Network/help/Add-AzVirtualNetworkSubnetConfig.md]: Please make this more human-readable e.g. "Id of NAT Gateway resource"

File: src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs [hunk 8]
```diff
@@ -61,13 +62,17 @@ public class AzureVirtualNetworkSubnetConfigBase : NetworkBaseCmdlet
...
         public PSRouteTable RouteTable { get; set; }
 
+        [GenericBreakingChange("Update Property Name", OldWay = "-ResourceId", NewWay = "-NatGatewayId")]
+        [Alias("NatGatewayId")]
         [Parameter(
             Mandatory = false,
             ValueFromPipelineByPropertyName = true,
             ParameterSetName = "SetByResourceId",
-            HelpMessage = "ResourceId")]
+            HelpMessage = "NatGatewayId")]
```
Comment (number213) [inline src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs]: Please make this more human-readable e.g. "Id of NAT Gateway resource"

File: src/Network/Network/help/New-AzVirtualNetworkSubnetConfig.md [hunk 9]
```diff
@@ -154,6 +179,21 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -ResourceId
+NatGatewayId
```
Comment (number213) [inline src/Network/Network/help/New-AzVirtualNetworkSubnetConfig.md]: Please make this more human-readable e.g. "Id of NAT Gateway resource"

File: src/Network/Network/help/Set-AzVirtualNetworkSubnetConfig.md [hunk 10]
```diff
@@ -128,6 +142,19 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -InputObject
```
Comment (number213) [inline src/Network/Network/help/Set-AzVirtualNetworkSubnetConfig.md]: Please add description for this parameter

File: src/Network/Network/help/Set-AzVirtualNetworkSubnetConfig.md [hunk 11]
```diff
@@ -173,6 +200,21 @@ Accept pipeline input: True (ByPropertyName)
 Accept wildcard characters: False
 ```
 
+### -ResourceId
+NatGatewayId
```
Comment (number213) [inline src/Network/Network/help/Set-AzVirtualNetworkSubnetConfig.md]: Please make this more human-readable e.g. "Id of NAT Gateway resource"

**Comment by number213** [comment]
CI failure is one failure in Websites' tests, seems unrelated to these changes

**Comment by niander** [comment]
@number213 We are taking a look into the websites failure. 
@khannarheams please address the review comments.

**Comment by number213** [comment]
Shouldn't this PR target `network-april`? @MikhailTryakhov

**Comment by MiYanni** [comment]
At this point, we'll have this PR target `network-april` as it is not ready for the current Az release.

**Comment by khannarheams** [comment]
@MiYanni created new PR for network-april branch https://github.com/Azure/azure-powershell/pull/9219

File: src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs [hunk 12]
```diff
@@ -61,18 +62,22 @@ public class AzureVirtualNetworkSubnetConfigBase : NetworkBaseCmdlet
             HelpMessage = "RouteTable")]
         public PSRouteTable RouteTable { get; set; }
 
+        [GenericBreakingChange("Update Property Name", OldWay = "-ResourceId", NewWay = "-NatGatewayId")]
```
Comment (MiYanni) [inline src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs]: We shouldn't have a breaking change attribute as we won't remove ResourceId since it is a standard parameter for Azure PowerShell cmdlets.

File: src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs [hunk 13]
```diff
@@ -61,18 +62,22 @@ public class AzureVirtualNetworkSubnetConfigBase : NetworkBaseCmdlet
...
+        [Alias("NatGatewayId")]
         [Parameter(
             Mandatory = false,
             ValueFromPipelineByPropertyName = true,
             ParameterSetName = "SetByResourceId",
-            HelpMessage = "ResourceId")]
+            HelpMessage = "Specifies the Id of NAT Gateway resource associated with the subnet configuration")]
         public string ResourceId { get; set; }
 
+        [GenericBreakingChange("Update Property Name", OldWay = "-InputObject", NewWay = "-NatGateway")]
```
Comment (MiYanni) [inline src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs]: We shouldn't have a breaking change attribute as we won't remove InputObject since it is a standard parameter for Azure PowerShell cmdlets.

File: src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs (hunk 12 above)
Comment (khannarheams) [reply src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs]: this was agreed upon in offline meeting with @niander  and @markcowl. We will retain resourceid on natgateway operations but can change parameter name when it is referenced from another operation. (Similar to RouteTableId here)

File: src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs (hunk 13 above)
Comment (khannarheams) [reply src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs]: this was agreed upon in offline meeting with @niander  and @markcowl. We will retain inputobject on natgateway operations but can change parameter name when it is referenced from another operation. (Similar to RouteTable here) @MiYanni

File: src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs (hunk 12 above)
Comment (MiYanni) [reply src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs]: Discussed with Niander. Looks good.

File: src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs (hunk 13 above)
Comment (MiYanni) [reply src/Network/Network/VirtualNetwork/Subnet/AzureVirtualNetworkSubnetConfigBase.cs]: Discussed with Niander. Looks good.

**Comment by khannarheams** [comment]
@MiYanni  updated the changelog
